### PR TITLE
[do not merge] segmentio - tweak bundledIds and include debug info

### PR DESCRIPTION
**Pull request by mericsson**
**What does this PR do?**

Include debug info in the segmentio event.

**Are there breaking changes in this PR?**

**Testing**

**Any background context you want to provide?**

**Is there parity with the server-side/android/iOS integration components (if applicable)?**

**Does this require a new integration setting? If so, please explain how the new setting works**

**Links to helpful docs and other external resources**

### CODE REVIEW

File: integrations/segmentio/package.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "@segment/analytics.js-integration-segmentio",
   "description": "The Segmentio analytics.js integration.",
-  "version": "4.4.1",
+  "version": "4.4.1-debug",
```
Comment (mericsson) [inline integrations/segmentio/package.json:4]: Will only deploy for source that we are debugging.

File: integrations/segmentio/lib/index.js [hunk 2]
```diff
@@ -316,11 +316,11 @@ Segment.prototype.normalize = function(message) {
   }
   if (this.options.addBundledMetadata) {
     var bundled = keys(this.analytics.Integrations);
-    var maybeBundledConfigIds = this.options.maybeBundledConfigIds
+    var maybeBundledConfigIds = this.options.maybeBundledConfigIds;
 
     // Generate a list of bundled config IDs using the intersection of
     // bundled destination names and maybe bundled config IDs.
-    var bundledConfigIds = []
+    var bundledIds = [];
```
Comment (mericsson) [inline integrations/segmentio/lib/index.js:323]: One theory I have is that `bundledConfigIds` was previously exposed in renderer snippet and may be stale and overriding the values here.

Comment (mericsson) [reply integrations/segmentio/lib/index.js:323]: https://github.com/segmentio/ajs-renderer/pull/256/files#diff-ca733fb7a822baffec24f4f426ccb07fce89f5b4d5856f848be4845d1d40e12bL61

File: integrations/segmentio/lib/index.js [hunk 3]
```diff
@@ -331,16 +331,16 @@ Segment.prototype.normalize = function(message) {
...
       }
     }
 
 
     msg._metadata = msg._metadata || {};
     msg._metadata.bundled = bundled;
     msg._metadata.unbundled = this.options.unbundledIntegrations;
-    msg._metadata.bundledIds = bundledConfigIds;
+    msg._metadata.bundledIds = bundledIds;
+    msg._metadata.maybeBundledConfigIds = maybeBundledConfigIds;
```
Comment (mericsson) [inline integrations/segmentio/lib/index.js:343]: Including here for debug purposes.

File: integrations/segmentio/lib/index.js [hunk 4]
```diff
@@ -331,16 +332,18 @@ Segment.prototype.normalize = function(message) {
       }
 
       for (var j = 0; j < maybeBundledConfigIds[name].length; j++) {
+        bundledIdsComputedAtRuntime = true;
```
Comment (mericsson) [inline integrations/segmentio/lib/index.js:335]: One theory I have is that bundledConfigIds was previously exposed in renderer snippet and may be stale and overriding the values here https://github.com/segmentio/ajs-renderer/pull/256/files#diff-ca733fb7a822baffec24f4f426ccb07fce89f5b4d5856f848be4845d1d40e12bL61

Comment (mericsson) [reply integrations/segmentio/lib/index.js:335]: This could happen if addBundledMetadata is false

File: integrations/segmentio/lib/index.js [hunk 5]
```diff
@@ -331,16 +332,19 @@ Segment.prototype.normalize = function(message) {
...
     }
 
-
     msg._metadata = msg._metadata || {};
     msg._metadata.bundled = bundled;
     msg._metadata.unbundled = this.options.unbundledIntegrations;
     msg._metadata.bundledIds = bundledConfigIds;
+    msg._metadata.maybeBundledConfigIds = maybeBundledConfigIds;
+    msg._metadata.bundledIdsComputedAtRuntime = bundledIdsComputedAtRuntime;
+    msg._metadata.addBundledMetadata = this.options.addBundledMetadata;
```
Comment (brennan) [inline integrations/segmentio/lib/index.js:347]: Maybe I missed this before - what's this option?

Comment (briemcnally) [inline integrations/segmentio/lib/index.js:347]: where is addBundledMetadata coming from?

Comment (brennan) [reply integrations/segmentio/lib/index.js:347]: Ah, I see here: https://github.com/segmentio/ajs-renderer/blob/63f438e1f1a5f2694f7a0d91f26f3e0a88287d4b/src/builder/settings-by-name.ts#L97

Comment (mericsson) [reply integrations/segmentio/lib/index.js:347]: https://github.com/segmentio/ajs-renderer/blob/63f438e1f1a5f2694f7a0d91f26f3e0a88287d4b/src/builder/settings-by-name.ts#L97

**Review by brennan — APPROVED** [review]

File: integrations/segmentio/lib/index.js [hunk 6]
```diff
@@ -314,14 +314,14 @@ Segment.prototype.normalize = function(message) {
...
     msg._metadata = { failedInitializations: failedInitializations };
   }
+  var bundledIdsComputedAtRuntime = false;
   if (this.options.addBundledMetadata) {
     var bundled = keys(this.analytics.Integrations);
     var maybeBundledConfigIds = this.options.maybeBundledConfigIds
 
     // Generate a list of bundled config IDs using the intersection of
     // bundled destination names and maybe bundled config IDs.
     var bundledConfigIds = []
```
Comment (mericsson) [inline integrations/segmentio/lib/index.js]: note: this is in the wrong scope

**Review by brennan — APPROVED** [review]

**Comment by mericsson** [comment]
Testing complete

**Pull request closed without merging**